types.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package raft
+
+// entryID uniquely identifies a raft log entry.
+//
+// Every entry is associated with a leadership term which issued this entry and
+// initially appended it to the log. There can only be one leader at any term,
+// and a leader never issues entries with the same index.
+type entryID struct {
+	term  uint64
+	index uint64
+}
+
+// logAppend is the metadata that describes a validated raft log append.
```

Comment:
I.e. the semantics of these appends are "from leader to log".
When this is generalized to a `logSlice`, the semantics will be "a sub-slice of the leader log at `logSlice.term`".

Reply:
The leader appends happen [here](https://github.com/pav-kv/raft/blob/d14e61bd82582490766aef1d4b13986fa2573a53/raft.go#L797-L813). That `raftLog.append` method should probably be [unified with](https://github.com/pav-kv/raft/blob/main/log.go#L107-L140) `maybeAppend` into one.

For a leader appending to its own log, the safety check on `Term` would trivially be true.

Reply:
Generalized the `logAppend` struct to be `logSlice` in this PR. It will be a central part of a few more clean-ups, so it's good lay the right foundation.